app/models/repository_host/github.rb

```diff
@@ -58,23 +58,31 @@ def commits_url(author = nil)
 
     def self.fetch_repo(id_or_name, token = nil)
       id_or_name = id_or_name.to_i if id_or_name.match(/\A\d+\Z/)
-      token ||= AuthToken.find_token(:v4).token
-
-      api_hash = AuthToken.fallback_client(token).repo(id_or_name, accept: "application/vnd.github.drax-preview+json,application/vnd.github.mercy-preview+json").to_hash
-
-      # disabling for now due to some permission issues in the GH API
-      # owner = api_hash.dig(:owner, :login)
-      # repository_name = api_hash[:name]
-      # graphql_client = AuthToken.new_v4_client(token)
-      # graphql_values = GraphqlRepositoryFieldsQuery.new(graphql_client).query(params: { owner: owner, repository_name: repository_name })
-      # graphql_hash = {
-      #   code_of_conduct_url: graphql_values.code_of_conduct_url,
-      #   contribution_guidelines_url: graphql_values.contribution_guidelines_url,
-      #   funding_urls: graphql_values.funding_urls,
-      #   security_policy_url: graphql_values.security_policy_url,
-      # }
-
-      RawUpstreamDataConverter.convert_from_github_api(api_hash)
+      required_scope = %w[repo public_repo]
+      token ||= AuthToken.find_token(:v4, required_scope: required_scope).token
+
+      github_client = AuthToken.fallback_client(token)
+      api_hash = github_client.repo(id_or_name, accept: "application/vnd.github.drax-preview+json,application/vnd.github.mercy-preview+json").to_hash
+
+      owner = api_hash.dig(:owner, :login)
+      repository_name = api_hash[:name]
+
+      # verify the token being used has the correct scopes for the document URLs API call
```

Comment:
I wrote it this way in case the token already contains the right scope. It's a difficult tracking down all the places where this method could be called and where the passed in token is decided on so this logic checks the existing token first before finding a new one. This also limits the usage of these tokens to just this API call so they don't become exhausted quickly making other API calls.